hibernate-core/src/main/java/org/hibernate/query/criteria/internal/QueryStructure.java

```diff
@@ -341,20 +358,48 @@ private void renderFromClause(StringBuilder jpaqlQuery, RenderingContext renderi
 	}
 
 	protected void renderWhereClause(StringBuilder jpaqlQuery, RenderingContext renderingContext) {
-		if ( getRestriction() == null ) {
+		final String correlationRestrictionWhereFragment = getCorrelationRestrictionsWhereFragment();
+		if ( getRestriction() == null && correlationRestrictionWhereFragment.isEmpty() ) {
 			return;
 		}
 
 		renderingContext.getClauseStack().push( Clause.WHERE );
 		try {
-			jpaqlQuery.append( " where " )
-					.append( ( (Renderable) getRestriction() ).render( renderingContext ) );
+			jpaqlQuery.append( " where " );
+			jpaqlQuery.append( correlationRestrictionWhereFragment );
+			if ( getRestriction() != null ) {
+				if ( !correlationRestrictionWhereFragment.isEmpty() ) {
+					jpaqlQuery.append( " and (" );
```

Comment:
```suggestion
					jpaqlQuery.append( " and " );
```

Reply:
My concern is without the parenthesis, there might be issue if the original SQL is 'or' compound predicate. For instance, the original SQL might be:
```
where alias0.name = 'haha' or alias0.field1 is null
```
then without parenthesis, the following erroneous SQL would be created:
```
where alias1 = alias0 and alias0.name = 'haha' or alias0.name is null
```
No such concern exists?

Reply:
updated as you suggest